src/cli/tree.rs

```diff
@@ -0,0 +1,401 @@
+        filtered_deps.retain(|p| regex.is_match(p));
+
+        if filtered_deps.is_empty() {
+            Err(miette::miette!(
+                "No top level dependencies matched the given regular expression"
+            ))?;
+        }
+    }
+
+    let mut visited_pkgs = Vec::new();
+    let direct_dep_count = filtered_deps.len();
+
+    for (index, pkg_name) in filtered_deps.iter().enumerate() {
+        visited_pkgs.push(pkg_name.to_owned());
+
+        let last = index == direct_dep_count - 1;
+        let symbol = if last {
+            UTF8_SYMBOLS.ell
+        } else {
+            UTF8_SYMBOLS.tee
+        };
+        if let Some(pkg) = dep_map.get(pkg_name) {
+            print_package(format!("{symbol} "), pkg, true, false);
+
+            let prefix = if last {
+                UTF8_SYMBOLS.empty
+            } else {
+                UTF8_SYMBOLS.down
+            };
+            print_dependency_leaf(
+                pkg,
+                format!("{} ", prefix),
+                dep_map,
+                &mut visited_pkgs,
+                direct_deps,
+            )
+        }
+    }
+    Ok(())
+}
+
+/// Recursively print top down dependency tree nodes
+fn print_dependency_leaf(
+    pkg: &Package,
+    prefix: String,
+    dep_map: &HashMap<String, Package>,
+    visited_pkgs: &mut Vec<String>,
+    direct_deps: &Vec<String>,
+) {
+    let dep_count = pkg.dependencies.len();
+    for (index, dep_name) in pkg.dependencies.iter().enumerate() {
+        let last = index == dep_count - 1;
+        let symbol = if last {
+            UTF8_SYMBOLS.ell
+        } else {
+            UTF8_SYMBOLS.tee
+        };
+
+        // skip virtual packages
+        if dep_name.starts_with("__") {
+            continue;
+        }
+
+        if let Some(dep) = dep_map.get(dep_name) {
+            let visited = visited_pkgs.contains(&dep.name);
+            visited_pkgs.push(dep.name.to_owned());
+
+            print_package(
+                format!("{prefix}{symbol} "),
+                dep,
+                direct_deps.contains(&dep.name),
+                visited,
+            );
+
+            if visited {
+                continue;
+            }
+
+            let new_prefix = if last {
+                format!("{}{} ", prefix, UTF8_SYMBOLS.empty)
+            } else {
+                format!("{}{} ", prefix, UTF8_SYMBOLS.down)
+            };
+            print_dependency_leaf(dep, new_prefix, dep_map, visited_pkgs, direct_deps);
+        }
+    }
+}
+
+/// Print package and style by attributes, like if are a direct dependency (name is green and bold),
+/// or by the source of the package (yellow version string for Conda, blue for PyPI).
+/// Packages that have already been visited and will not be recursed into again are
+/// marked with a star (*).
+fn print_package(prefix: String, package: &Package, direct: bool, visited: bool) {
+    println!(
+        "{}{} v{} {}",
```

Comment:
I think the `v` should also be colored!

Reply:
I'll tweak that, though maybe not till I'm supposed to be sitting in a workshop this afternoon and trying to avoid escaping.

Reply:
I would actually just remove the `v` as it isn't an actual part of the version. And `pixi list` also doesn't have it.

Reply:
The `v` is gone.